pkg/backup/compaction_test.go

```diff
@@ -556,6 +548,187 @@ func TestBlockConcurrentScheduledCompactions(t *testing.T) {
 	require.Equal(t, 2, numCompactions)
 }
 
+func TestBackupCompactionExecLocality(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	skip.UnderRace(t, "too slow")
+
+	args := base.TestClusterArgs{
+		ServerArgs: base.TestServerArgs{
+			DefaultTestTenant: base.TestDoesNotWorkWithSecondaryTenantsButWeDontKnowWhyYet(142798),
+		},
+		ServerArgsPerNode: map[int]base.TestServerArgs{
+			0: {
+				ExternalIODir: "/west0",
+				Locality: roachpb.Locality{Tiers: []roachpb.Tier{
+					{Key: "tier", Value: "0"},
+					{Key: "region", Value: "west"},
+				}},
+			},
+			1: {
+				ExternalIODir: "/west1",
+				Locality: roachpb.Locality{Tiers: []roachpb.Tier{
+					{Key: "tier", Value: "1"},
+					{Key: "region", Value: "west"},
+				}},
+			},
+			2: {
+				ExternalIODir: "/east0",
+				Locality: roachpb.Locality{Tiers: []roachpb.Tier{
+					{Key: "tier", Value: "0"},
+					{Key: "region", Value: "east"},
+				}},
+			},
+			3: {
+				ExternalIODir: "/east1",
+				Locality: roachpb.Locality{Tiers: []roachpb.Tier{
+					{Key: "tier", Value: "1"},
+					{Key: "region", Value: "east"},
+				}},
+			},
+		},
+	}
+	const numAccounts = 1000
+	tc, db, _, cleanup := backupRestoreTestSetupWithParams(t, 4, numAccounts, InitManualReplication, args)
+	defer cleanup()
+
+	west1Node, east0Node := sqlutils.MakeSQLRunner(tc.Conns[1]), sqlutils.MakeSQLRunner(tc.Conns[2])
+
+	targets := "DATABASE data"
+
+	// bring all node dirs up to an identical starting point
```

Comment:
At Cockroach, we tend to be a bit more formal with our comments as far as grammar goes, so they should start with a capital letter. You can read more on our [comment guidelines](https://cockroachlabs.atlassian.net/wiki/spaces/CRDB/pages/914260289/Code+commenting+guidelines) here.

Actually that section of our docs has a lot of good notes on our general coding/operational guidelines so it'd be a good read during your downtime.